packages/strapi-admin/admin/src/containers/MarketplacePage/index.js

```diff
@@ -10,20 +10,31 @@ import Wrapper from './Wrapper';
 
 const MarketPlacePage = () => {
   const history = useHistory();
-  const { autoReload, currentEnvironment, formatMessage, plugins } = useGlobalContext();
+  const { autoReload, emitEvent, currentEnvironment, formatMessage, plugins } = useGlobalContext();
   const { error, isLoading, data } = useFetchPluginsFromMarketPlace();
 
+  const memoizedEmit = useCallback(() => {
+    emitEvent('didGoToMarketplace');
+  }, [emitEvent]);
+
+  useEffect(() => {
+    memoizedEmit();
+  }, [memoizedEmit]);
+
```

Comment:
@soupette Is this the correct way to do this?  Originally I thought if I just added an empty dependency array to the useEffect it would only fire on first render, but eslint wanted the `emitEvent` function in the dependency array.

Reply:
the best way to remove the emit from the dependency array is to use a ref:

```js
const emitEventRef = useRef(emitEvent);

useEffect(() => {
  emitEventRef.current('didGotToMarketplace');
}, [])
```